pkg/services/sqlstore/bulk.go

```diff
@@ -0,0 +1,59 @@
+package sqlstore
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+const DefaultBatchSize = 1000
+
+type BulkOpSettings struct {
+	BatchSize int
+}
+
+func NativeSettingsForDialect(d migrator.Dialect) BulkOpSettings {
```

Comment:
I'd have liked to hide this detail, but I really wanted to make this work at the session level rather than the store level. This is so that the caller has complete control over where this utility is used - we will respect transactions or any other context on the same session.

Unfortunately, the exported API for session doesn't seem to offer a way to obtain the dialect of its parent store. It seemed risky to add such a link here.